src/Classes/Helicoptere.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+class Helicoptere {
+
+    //coordonnees en x y z
+    public $x,$y,$z;
+    //direction vers laquelle est tournee l'helico
+    public $direction;
+
+    function __construct($posx,$posy) {
+        $this->$x=$posx;
+        $this->$y=$posy;
+        $this->$direction=0;
+    }
+
+    function seDeplacer($distance) {
+        $angle=($direction % 360);
+        switch ($angle) {
+            //si tout droit avance en y
+            case 0:
+                if(($this->$y+$distance)>=12){
+                    $this->$y=12;
+                }else{
+                    $this->$y+=$distance;
+                }
+            break;
+            //si vers la droite avance en x
+            case 90:
+                if(($this->$x+$distance)>=20){
+                    $this->$x=20;
+                }else{
+                    $this->$x+=$distance;
+                }
+            break;
+            //si vers le bas recule en y
+            case 180:
+                if(($this->$y-$distance)<=0){
+                    $this->$y=0;
+                }else{
+                    $this->$y-=$distance;
+                }
+            break;
+            //si versla gauche recule en x
+            case 270:
+                if(($this->$x-$distance)<=0){
+                    $this->$x=0;
+                }else{
+                    $this->$x-=$distance;
+                }
+            break;
+            //si diagonale haut droite avance y et x
+            case 45:
+                //il faut vérifier si on peut avancer àchaque fois
+                $resDistance=$distance;
+                while($distance>0){
```

Comment:
Usage of a while loop is usually a sign of a code smell, can you do anyhow else?